### PR TITLE
Fix "Cannot charge a customer that has no active card" error

### PR DESCRIPTION
**Pull request by orlando**
**What:** Fix "Cannot charge a customer that has no active card" error

**Why:** Fixes https://app.asana.com/0/1168997577035609/1199605570263293

**How:**

Replicate test via integration testing. There's an issue with the `stripe_mock` library, where this test case wasn't failing. It took me a while to realize what the issue was. For the integration test with cypress, we let the server make requests to the Stripe test API so it tests the behaviour more closely to the live integration.

**Media:**

**replicate failing edge case**
image

### CODE REVIEW

File: app/jobs/subscription_payment_job.rb [hunk 1]
```diff
@@ -26,7 +26,11 @@ def perform(subscription)
 
   def create_charge(subscription)
     user = subscription.user
-    customer = find_stripe_customer(user)
+    customer = user.find_stripe_customer
+
+    unless customer
+      # this shouldn't happen, but we need to handle this case
```
Comment (duranmla) [inline app/jobs/subscription_payment_job.rb]: If we need this to be handle do we have a ticket for this?

File: cypress/integration/membership.spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      cy.contains('$10').click()
+      cy.get('button')
+        .contains('PAY MONTHLY DUES', { matchCase: false })
+        .click()
+
+      // fill address
+      cy.contains('PAYING $10', { matchCase: false }).click()
+
+      const zipCode = faker.address.zipCode()
+      cy.get("input[name='address']").type(faker.address.streetAddress())
```
Comment (duranmla) [inline cypress/integration/membership.spec.js:33]: If you use instead https://testing-library.com/docs/cypress-testing-library/intro/ we will have a unified "language" on the test side as you will be using rtl queries instead.

_Highly opinionated comment, feel free to proceed as you believe convenient_

File: cypress/support/on-rails.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+// CypressOnRails: dont remove these command
+Cypress.Commands.add('appCommands', function (body) {
+  cy.log('APP: ' + JSON.stringify(body))
```
Comment (duranmla) [inline cypress/support/on-rails.js:3]: It's ok to have this one?

File: cypress/support/on-rails.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  cy.visit('__cypress__/force_login', {
+    method: 'POST',
+    body: { email: details.email, redirect_to: details.redirect_to }
+  })
+})
+
+// CypressOnRails: end
+
+// The next is optional
+// beforeEach(() => {
```
Comment (duranmla) [inline cypress/support/on-rails.js]: Maybe no need the code commented here, isn't?

**Review by duranmla — APPROVED** [review]
👌

File: app/jobs/subscription_payment_job.rb (hunk 1 above)
Comment (orlando) [reply app/jobs/subscription_payment_job.rb]: I'll add some handling for this

File: cypress/integration/membership.spec.js (hunk 2 above)
Comment (orlando) [reply cypress/integration/membership.spec.js:33]: Yeah I think we can add this later.

File: cypress/support/on-rails.js (hunk 3 above)
Comment (orlando) [reply cypress/support/on-rails.js:3]: mmm, I'm not sure, this comes with CypressOnRails. I don't know if it's being used by it internally tbh.

**Comment by orlando** [comment]
Merging. Thanks for taking a look @duranmla
